File: challenges/palindromes/python/palindrome.py
import re


def palindrome(word):
    chars = [char for char in str(word).lower() if re.match('[a-z0-9]', char)]
    while len(chars) > 1:
        if chars[0] != chars[-1]:
            return False
        else:
            chars.pop(0)
            chars.pop(-1)
    return True

File: challenges/palindromes/js/palindrome.js
const palindrome = (word) => {
    let string = word.toString().toLowerCase()
    const chars = string.split('').filter((char) => {
        return char.match('[a-z0-9]')
    })

    while (chars.length > 1) {
        if (chars[0] !== chars[chars.length - 1]) {
            return false
        } else {
            chars.pop()
            chars.shift()
        }
    }
    return true
};

module.exports = { palindrome }